pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyTestUtils.java

```diff
@@ -79,4 +80,26 @@ public LatchControlledRequestHandlerFactory(LatchControlledRequestHandler reques
       return _requestHandler;
     }
   }
+
+  public static void waitForServerStarted(NettyTCPServer server)
+      throws TimeoutException {
+    // Wait for at most 1 second for server to start
+    long endTime = System.currentTimeMillis() + 1000L;
+    while (System.currentTimeMillis() < endTime) {
+      if (server.isStarted()) {
```

Comment:
We are running on multi-core CPU so that should be fine (actually that's the reason why we need volatile :P).